src/tribler-core/tribler_core/components/upgrade.py

```diff
@@ -22,8 +23,9 @@ async def run(self):
             trustchain_keypair=master_key_component.keypair,
             notifier=notifier)
 
-        rest_component = await self.require_component(RESTComponent)
-        rest_component.rest_manager.get_endpoint('upgrader').upgrader = self.upgrader
-        rest_component.rest_manager.get_endpoint('state').readable_status = STATE_UPGRADING_READABLE
-
+        await self.init_endpoints(['upgrader'], [('upgrader', self.upgrader)])
+        await self.set_readable_status(STATE_UPGRADING_READABLE)
         await self.upgrader.run()
+
+    async def shutdown(self):
+        self.release_endpoints()
```

Comment:
I would expect `release_endpoints` to be called automatically on every call to `shutdown` of `RestfulComponent`. That's why we subclass stuff, right?

Reply:
Done